### DISCUSSION TITLE
Output only one file not all folder in output folder

### DISCUSSION THREAD

**Discussion by daidaiV5** [nextflow-io/nextflow#2070] [Q&A]
Hello:

I would like to output only one file not all folder in output folder

for example :

    process a {
     publishDir path: './resultat' 
     echo true

     output:
     file("a/**/2.txt") into found

    """
    mkdir -p a/a b/b c/c
    touch a/1.txt
    touch b/1.txt
    touch c/1.txt
    touch a/a/2.txt
    touch b/b/2.txt
    touch c/c/2.txt
    """   }

Now I output a folder in folder ‘resultat’: ./resultat/a/a/2.txt, but I only want to output '2.txt' in the folder ‘resultat’ : ./resultat/2.txt
How to do that?

Thanks for taking the time to read.

**Comment by pditommaso** [comment]
use `output: file('a/a/2.txt')` instead of `**`

**Reply by daidaiV5** [reply]
thanks for your reply, but it's not my question, I can output the file 'a/a/2.txt', but I would like only a file not a folder. because even I use output: file('a/a/2.txt'), The results are: '/resultat/a/a/2.txt', but I only want to output '2.txt' in the folder ‘resultat’ : it should like ./resultat/2.txt

**Reply by pditommaso** [reply]
This may help https://github.com/nextflow-io/patterns/blob/master/docs/publish-rename-outputs.adoc

**Comment by daidaiV5** [comment]
thanks a lot, I see in link is 'Save outputs in a sub-directory', but I would like to bring the file out of sub-directory and store in directory not sub-directory, if I found the solution, I will complete here.
